docs/example-confluence/src/knowledge.ts

```diff
@@ -0,0 +1,195 @@
+import { createKnowledgeDescriptor } from "@cognigy/extension-tools";
+import { ConfluenceDataParser } from "./confluenceParser";
+const { CharacterTextSplitter } = require("@langchain/textsplitters");
+
+const MAX_CHUNK_SIZE = 2000; // Define a maximum chunk size in characters
+
+/**
+ * Headings less than and equal to this level create new chunks
+ * (H1,H2 → new chunks; H3+ → continue chunk)
+ */
+const TARGET_HEADING_LEVEL = 2;
+
+export const confluenceImport = createKnowledgeDescriptor({
+    type: "confluenceKnowledgeConnector",
+    label: "Confluence",
+    summary: "Imports text under each heading from all pages in a Confluence directory",
+    fields: [
+        {
+            key: "connection",
+            label: "Connect to Confluence",
+            type: "connection",
+            params:{
+                connectionType: "api-key",
+                required: "true"}
+        },
+        {
+            key: "confluenceUrl",
+            label: "Confluence URL",
+            type: "text",
+            params: { required: true },
+            description: "The URL of the Confluence page or folder to import."
+        },
+        {
+            key: "descendants",
+            label: "Extract Descendants",
+            type: "toggle",
+            params: { required: true},
+            defaultValue: true,
+            description: "Extract all child pages under the parent page. For folders, descendants are always extracted."
+        },
+        {
+            key: "source_tag",
+            label: "Source Tags",
+            type: "chipInput",
+            params: { required: false },
+            defaultValue: ["confluence"],
+            description: "Source tags can be used to filter the search scope from the Flow. Press ENTER to add a Source Tag.",
+        }
+    ],
```

Comment:
The fields should be defined as const, otherwise the framework does not provide typing support for configuration values

```suggestion
    ] as const,
```